source/extensions/filters/listener/local_ratelimit/local_ratelimit.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include "envoy/event/file_event.h"
+#include "envoy/event/timer.h"
+#include "envoy/extensions/filters/listener/local_ratelimit/v3/local_ratelimit.pb.h"
+#include "envoy/network/filter.h"
+#include "envoy/runtime/runtime.h"
+#include "envoy/stats/scope.h"
+#include "envoy/stats/stats_macros.h"
+
+#include "source/common/common/logger.h"
+#include "source/common/runtime/runtime_protos.h"
+#include "source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace LocalRateLimit {
+
+/**
+ * All local rate limit stats. @see stats_macros.h
+ */
+#define ALL_LOCAL_RATE_LIMIT_STATS(COUNTER) COUNTER(rate_limited)
+
+/**
+ * Struct definition for all local rate limit stats. @see stats_macros.h
+ */
+struct LocalRateLimitStats {
+  ALL_LOCAL_RATE_LIMIT_STATS(GENERATE_COUNTER_STRUCT)
+};
+
+using LocalRateLimiterImplPtr =
+    std::unique_ptr<Filters::Common::LocalRateLimit::LocalRateLimiterImpl>;
+
+/**
+ * Configuration shared across all connections. Must be thread safe.
+ */
+class FilterConfig : Logger::Loggable<Logger::Id::filter> {
+public:
+  FilterConfig(
+      const envoy::extensions::filters::listener::local_ratelimit::v3::LocalRateLimit& proto_config,
+      Event::Dispatcher& dispatcher, Stats::Scope& scope, Runtime::Loader& runtime);
+
+  ~FilterConfig() = default;
+
+  bool canCreateConnection();
+  bool enabled() { return enabled_.enabled(); }
+  const LocalRateLimitStats& stats() const { return stats_; }
+
+private:
+  static LocalRateLimitStats generateStats(const std::string& prefix, Stats::Scope& scope);
+
+  const Runtime::FeatureFlag enabled_;
+  const LocalRateLimitStats stats_;
+  LocalRateLimiterImplPtr rate_limiter_;
+};
+
+using FilterConfigSharedPtr = std::shared_ptr<FilterConfig>;
+
+/**
+ * Local Rate Limit listener filter.
+ */
+class Filter : public Network::ListenerFilter, Logger::Loggable<Logger::Id::filter> {
+public:
+  Filter(const FilterConfigSharedPtr& config) : config_(config) {}
+
+  // Network::ListenerFilter
+  Network::FilterStatus onAccept(Network::ListenerFilterCallbacks& cb) override;
+  Network::FilterStatus onData(Network::ListenerFilterBuffer&) override {
```

Comment:
I am a bit confused, why we  override `onData` and `maxReadBytes` here as this filter inherits from `Network::ListenerFilter` which only has one `onAccept()`

Reply:
since we don't have a pass through listener filter to inherit iirc. i could have one but that's should be a separated PR.

Reply:
i.e., all method in ListenerFilter are pure abstract and needs to be override

Reply:
Sorry, I am trying github vscode codespace out.  Somehow, it only shows `virtual FilterStatus onAccept(ListenerFilterCallbacks& cb) PURE;` (same to all other inherited classes).

I checked the latests code. My comment should be ignored.

Reply:
ah, I create the codespace based off my fork repo that is [6590 commits behind](https://github.com/tyxia/envoy/compare/main...envoyproxy:envoy:main)... 
Sorry, please ignore this out-of-date comment